include/xtensor/xfixed.hpp

```diff
@@ -950,4 +950,32 @@ namespace xt
     }
 }
 
+/******************************
+ * std::tuple_size extensions *
+ ******************************/
+
+#if defined(__clang__)
+    # pragma clang diagnostic push
+    # pragma clang diagnostic ignored "-Wmismatched-tags"
+#endif
+
+namespace std
```

Comment:
This specialization was done in the same style as here: https://github.com/xtensor-stack/xtensor/blob/308458574baa95b0dabb981a4bae97fa29ff673f/include/xtensor/xstorage.hpp#L1915

It also contains a comment about tuple_size being a class in c++ standard, I can just copy the comment here, or update also the xstorage version.

Reply:
My initial comment was based on the standard definition of [tuple_size](http://eel.is/c++draft/tuple.helper#lib:tuple_size). However, it appears that th standard uses `class tuple_size` in other parts of the spec, and is therefore inconsistent. libc++ is consistent and uses `class` everywhere, so we can keep it. See this [discussion](https://github.com/llvm/llvm-project/issues/39218#issuecomment-981006038) for more detail about this issue.

The specializations in `xstorage.hpp` should be updated to avoid opening the std namespace, though.